### PR TITLE
Implement perunnode-cli

### LINKED ISSUE #99: Reference implementation for a perun node client application
#### Motivation / Context

Implement an interactive command line application that can be used as a
1. Reference implementation for using the client api (from gRPC generated code) to interact with a perun node.
2. Tool for new users to try out the perun node without writing any code.

#### Description

The app should provide the following commands and the corresponding options:
1. `connect`: connect with an instance of perun node and start an interactive shell.
  a.`--configFile`: path to the session config file for the user. The file should be present on the machine running the node instance and this is is the path on that machine and not on the machine running the client instance.
  b. `--peruNodeURL`: URL of the perun-node to connect with.
  d. `--chainURL`: (Optional) Address of the blockchain node.
  e. `--adjudicator`:  (Optional) Address of the Adjudicator contract.
  f. `--asset`:  (Optional) Address of the Asset Holder contract.
If the optional parameters are specified, these values will override the values contained in the config file.

The app should connect to a running instance of perun node as specified in the conifg file via `gRPC`. Within the interactive shell, it should
1. Provides commands for using the payment api of the exposed by the node.
2. Receive and print notifications that a user has subscribed for.

The code to receive and print the subscription should run as a go-routine. When the app is closed by exiting the interactive shell, the subscriptions (if present) should be unsubscribed.

##### Relates to

Depends on #98 

#### Testing

1. Manual tests

#### Thread Safety

Not required, only one of the command will be invoked by the user at a given point of time.

### PR DESCRIPTION
**Pull request by manoranjith**
#### Description

- perunnode-cli is an interactive CLI application to connect with
perunnode and use the payment channel API. It uses the grpc client stubs
from api/grpc package to connect with grpc payment channel API server
running in the node.

- Besides commands to access the APIs on node, the app also has a chain
command that can directly interact with blockchain to deploy contracts.

- The app is designed such that valid values for almost all of the
positional arguments are suggested by autocomplete engine, that can be
invoked by pressing tab.

- Also added a readme file in cmd/perunnodecli directory with a tutorial
to try out the perunnode and
perunnode-cli.

##### Category

Implementation Task

##### Relevant issue

Closes #99.

#### Testing

Included  a tutorial for perunnode-cli at `cmd/perunnodecli/Readme.md`. Everything should work as described in that document. 

#### Checklist 

- [x] Name is added to the NOTICE file, if it is not present already.
- [x] Changes are rebased onto the target branch.

### CODE REVIEW

File: cmd/perunnodecli/README.md [hunk 1]
```diff
@@ -0,0 +1,280 @@
...
+
+- Node: node.yaml file.
+- Session: Two directories (alice and bob) each containing session.yaml file,
+  contacts.yaml file and keystore directory with keys corresponding to the
+  on-chain and off-chain accounts.
+
+4. Start the perunnode:
+
+```
+./perunnode generate
```
Comment (anagha-ks) [inline cmd/perunnodecli/README.md]:
```suggestion
./perunnode run
```

File: cmd/perunnodecli/README.md [hunk 2]
```diff
@@ -0,0 +1,280 @@
...
+This will start the perunnode using the config file located at default path
+`./node.yaml` that was generated in step 3. You will see a line
+"Serving payment channel API via grpc at port :50001" with a blinking cursor.
+Leave this running in this terminal.
+
+Now all the pre-requisites for `perunnode-cli` are setup.
+
+### Opening a session, opening channel within it, making payments & closing it
+
+1. Open two new terminals side by side, once each for alice and bob roles
```
Comment (anagha-ks) [inline cmd/perunnodecli/README.md]:
```suggestion
1. Open two new terminals side by side, one each for alice and bob roles
```

**Review by anagha-ks — CHANGES_REQUESTED** [review]

File: cmd/perunnodecli/README.md [hunk 3]
```diff
@@ -0,0 +1,280 @@
...
+
+- (a) [Alice] Send a request to open a channel with bob:
+
+```
+# [Alice]
+channel send-opening-request bob 1 2
+```
+
+- (b) [Bob] Receives a channel opening request notification that includes
+  request ID. Accept it:
```
Comment (anagha-ks) [inline cmd/perunnodecli/README.md]:
```suggestion
  request ID. Type the command to accept the channel opening request directly after receiving the notification:
```

**Review by anagha-ks — CHANGES_REQUESTED** [review]

File: cmd/perunnodecli/README.md (hunk 3 above)
Comment (manoranjith) [reply cmd/perunnodecli/README.md]: Signed-off-by: Manoranjith <ponraj.manoranjitha@in.bosch.com>

**Review by anagha-ks — APPROVED** [review]

**Pull request merged**